plugins/out_stackdriver/stackdriver.c

```diff
@@ -277,6 +277,119 @@ static char *get_google_token(struct flb_stackdriver *ctx)
     return ctx->o->access_token;
 }
 
+/*
+* parse_labels
+* - Iterate throught the original payload (obj) and find out the entry that matches the labels_key 
+* - Convert all labels under labels_key to root-level `labels` field
+*/
+static msgpack_object *parse_labels(struct flb_stackdriver *ctx, msgpack_object *obj)
+{
+    int i;
+    int len;
+    msgpack_object_kv *kv = NULL;
+
+    if (!obj || obj->type != MSGPACK_OBJECT_MAP) {
+        return NULL;
+    }
+
+    len = flb_sds_len(ctx->labels_key);
+    for (i = 0; i < obj->via.map.size; i++) {
+        kv = &obj->via.map.ptr[i];
+        if (kv->key.via.str.size == len &&
+            strncasecmp(kv->key.via.str.ptr, ctx->labels_key, len) == 0) {
+            if (kv->val.type != MSGPACK_OBJECT_MAP) {
+                flb_plg_warn(ctx->ins, "the type of labels should be map");
+            }
+            return &kv->val;
+        }
+    }
+
+    flb_plg_debug(ctx->ins, "labels_key [%s] not found in the payload", 
+                  ctx->labels_key);
+    return NULL;
+}
+
+/* using packer mp_pck to pack obj except some special elements */
+static int pack_object_except(msgpack_object *obj, msgpack_packer mp_pck, 
```

Comment:
Most (all?) of the msgpack functions have the packer first, followed by the things that are being added to it. I'd recommend following the same pattern here, i.e.: `static int pack_object_except(msgpack_packer, msgpack_object, ctx)`

Reply:
this function has been replaced by https://github.com/fluent/fluent-bit/blob/master/plugins/out_stackdriver/stackdriver.c#L699